ui/pages/permissions-connect/connect-page/connect-page.tsx

```diff
@@ -92,10 +93,25 @@ export const ConnectPage: React.FC<ConnectPageProps> = ({
       ),
     [networkConfigurations],
   );
+
+  // By default, if a non test network is the globally selected network. We will only show non test networks as default selected.
+  const currentlySelectedNetwork = useSelector(getMultichainNetwork);
+  const currentlySelectedNetworkChainId =
+    currentlySelectedNetwork.network.chainId;
+  // If globally selected network is a test network, include that in the default selcted networks for connection request
+  const selectedTestNetwork = testNetworks.filter(
+    (network: { chainId: string }) =>
+      network.chainId === currentlySelectedNetworkChainId,
+  );
```

Comment:
there should only be one network that matches the chainId, does that sound right to you? We may want to use `.find()` here to make that clearer if so

Reply:
Updated https://github.com/MetaMask/metamask-extension/pull/27980/commits/21f3db86112e79d5c2941ec6f7a94f8179b9fcd1